modules/contentpool_remote_register/src/RegistrationPullManager.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Drupal\contentpool_remote_register;
+
+use Drupal\contentpool_remote_register\Entity\RemoteRegistration;
+use Drupal\Core\Entity\EntityTypeManagerInterface;
+use Drupal\Core\Messenger\MessengerInterface;
+use Drupal\Core\StringTranslation\StringTranslationTrait;
+use Drupal\multiversion\Entity\WorkspaceInterface;
+use Drupal\multiversion\Workspace\ConflictTrackerInterface;
+use Drupal\relaxed\SensitiveDataTransformer;
+use Drupal\workspace\Entity\WorkspacePointer;
+use Drupal\workspace\ReplicatorInterface;
+use GuzzleHttp\ClientInterface;
+use GuzzleHttp\RequestOptions;
+use Symfony\Component\Serializer\Serializer;
+
+/**
+ * Helper class to get training references and backreferences.
+ */
+class RegistrationPullManager implements RegistrationPullManagerInterface {
+
+  use StringTranslationTrait;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManager
+   */
+  protected $entityTypeManager;
+
+  /**
+   * The serializer.
+   *
+   * @var \Symfony\Component\Serializer\Serializer
+   */
+  protected $serializer;
+
+  /**
+   * The Guzzle HTTP client.
+   *
+   * @var \GuzzleHttp\Client
+   */
+  protected $httpClient;
+
+  /**
+   * The sensitive data transformer.
+   *
+   * @var \Drupal\relaxed\SensitiveDataTransformer
+   */
+  protected $sensitiveDataTransformer;
+
+  /**
+   * The Messenger service.
+   *
+   * @var \Drupal\Core\Messenger\MessengerInterface
+   */
+  protected $messenger;
+
+  /**
+   * Constructs a RemoteAutopullManager object.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, Serializer $serializer, ClientInterface $http_client, SensitiveDataTransformer $sensitive_data_transformer, MessengerInterface $messenger) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->httpClient = $http_client;
+    $this->sensitiveDataTransformer = $sensitive_data_transformer;
+    $this->messenger = $messenger;
+    $this->serializer = $serializer;
+  }
+
+  /**
+   * @inheritdoc
+   */
+  public function pullFromRemoteRegistrations() {
```

Comment:
this is a very confusing name based upon internal implemenation details. let's call it pushToRegisteredRemotes() please.